### PR TITLE
Fix 32-bit RLE decode and add extra test cases

### PR DESCRIPTION
**Pull request by zmb3**
Captured some real bitmap data and found a few edge cases with 32bpp decoding to fix.

Note: this also sets the alpha plane to fully opaque (0xFF), making the adjustment on the Teleport side unnecessary.

### CODE REVIEW

File: src/codec/rle.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ fn process_plane(
 
     let mut this_line: u32;
     let mut last_line: u32 = 0;
+    let mut raw = 0i8;
```
Comment (LKozlowski) [inline src/codec/rle.rs:16]: Does it need to be `i8` type?  The first time you set it you read `u8` value into it and cast it to `i8`, and then later when you use it you cast that `i8` type into `u8`. Maybe `u8` type would work from the start and then you could drop those type conversions?

**Review by LKozlowski — APPROVED** [review]

File: src/codec/rle.rs (hunk 1 above)
Comment (zmb3) [reply src/codec/rle.rs]: Yep, good catch. Fixed.
